### PR TITLE
CORE-20721: Add avro definitions to support new Group RestResource

### PR DESCRIPTION
**Pull request by Tom-Fitzpatrick**
This adds new avro schemas to support the new RBAC Group Rest Resource being added in https://github.com/corda/corda-runtime-os/pull/6233.

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/permissions/management/PermissionManagementRequest.avsc [hunk 1]
```diff
@@ -24,6 +24,11 @@
         "net.corda.data.permissions.management.user.AddRoleToUserRequest",
         "net.corda.data.permissions.management.user.RemoveRoleFromUserRequest",
         "net.corda.data.permissions.management.user.DeleteUserRequest",
+        "net.corda.data.permissions.management.group.AddRoleToGroupRequest",
+        "net.corda.data.permissions.management.group.ChangeGroupParentIdRequest",
```
Comment (vkolomeyko) [inline data/avro-schema/src/main/resources/avro/net/corda/data/permissions/management/PermissionManagementRequest.avsc:23]: While we are on it, shall we also introduce: `ChangeUserParentGroupIdRequest` ?

Comment (Tom-Fitzpatrick) [reply data/avro-schema/src/main/resources/avro/net/corda/data/permissions/management/PermissionManagementRequest.avsc:23]: Ah, good call. Added.

**Review by vkolomeyko — DISMISSED** [review]
LGTM

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**